hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -705,6 +723,23 @@ func deleteMachineDeployment(ctx context.Context, c client.Client, md *capiv1.Ma
 	return nil
 }
 
+func pauseMachineDeployment(ctx context.Context, c client.Client, md *capiv1.MachineDeployment) error {
+	err := c.Get(ctx, client.ObjectKeyFromObject(md), md)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("error getting MachineDeployment: %w", err)
+	}
+	if md.Annotations == nil {
+		md.Annotations = make(map[string]string)
+	}
+	//FIXME: In future we may want to use the spec field instead
+	// https://github.com/kubernetes-sigs/cluster-api/issues/6966
+	md.Annotations[capiv1.PausedAnnotation] = "true"
+	return c.Update(ctx, md)
```

Comment:
Again this is ok for consistency but future todo: use r.CreateOrUpdate / patch here.

Reply:
Ack - if a NodePool is created in paused state, we will call this before any MachineDeployment gets created (because we never reached the reconcile code that creates it) - so I don't think CreateOrUpdate would be appropriate, but I can switch this to patch if you prefer :+1:

Reply:
I left this as-is for now, let me know if you're prefer it moved to `patchHelper.Patch`